mlflow/types/utils.py

```diff
@@ -153,6 +153,8 @@ def _infer_schema(data: Any) -> Schema:
         isinstance(data, list)
         and all(isinstance(element, dict) for element in data)
         and all(isinstance(key, str) for d in data for key in d)
+        # NB: We allow both str and List[str] as values in the dictionary
+        # e.g. [{'output': 'some sentence', 'ids': ['id1', 'id2']}]
```

Comment:
```suggestion
        # e.g. [{'output': 'some sentence'}] or [{'ids': ['id1', 'id2']}]
```

Reply:
We accept a combination of both in the same dictionary.

Reply:
Makes sense :)